olive/passes/onnx/rtn_quantization.py

```diff
@@ -101,22 +100,29 @@ def _quantize_model(
                 if (node.op_type == OpType.MatMul and not node.inputs[1].is_initializer()) or (
                     node.op_type == OpType.Gather and not node.inputs[0].is_initializer()
                 ):
+                    logger.debug("skip to quantize %s as it has no initializer", node_name)
                     continue
 
-                quantized_node = self.quantize(node)
+                if node.op_type == OpType.Gather and self.bits != 4:
+                    logger.warning("Gather only supports 4-bit quantization.")
+                    continue
+
+                quantized_node, initializer_graph = self._quantize(node)
 
                 if quantized_node.op_type in (OpType.MatMulNBits, OpType.GatherBlockQuantized):
+                    registered = set()
                     for input_value in quantized_node.inputs:
-                        if input_value.const_value is not None:
-                            node.graph.register_initializer(input_value)
+                        if input_value.const_value is not None and input_value.name not in registered:
+                            initializer_graph.register_initializer(input_value)
+                            registered.add(input_value.name)
```

Comment:
```suggestion
                        if input_value.const_value is not None and input_value.name not in registered:
                            initializer_graph.register_initializer(input_value)
                            registered.add(input_value.name)
                        else:
                            ir.convenience.replace_all_uses_with(input_value, registered[name])
```

What should happen when input_value.name is registered? Should we then connect nodes with the existing value?